src/repo.c

```diff
@@ -137,14 +137,25 @@ index_diff(struct index_diff *diff, bool untracked, bool count_all)
 	};
 	struct io io;
 	struct buffer buf;
+	unsigned skip_count = 0;
 	bool ok = true;
 
 	memset(diff, 0, sizeof(*diff));
 
 	if (!io_run(&io, IO_RD, repo.cdup, NULL, status_argv))
 		return false;
 
-	while (io_get(&io, &buf, 0, true) && (ok = buf.size > 3)) {
+	while (io_get(&io, &buf, 0, true) && ((ok = buf.size > 3) || skip_count)) {
+
+		if (skip_count > 0){
+			skip_count--;
+			continue;
+		}
+
+		/* Skip source filename in rename */
+		if (buf.data[0] == 'R')
```

Comment:
This will still get incremented at line 152 as it did before. The next iteration where it reads in the 2nd filename will skip. Originally I used a counter because I thought it'd hit two filenames. Would it be more clear to use a flag like 'skip_next' or 'last_was_rename'?